위클리_미션/signin.js

```diff
@@ -0,0 +1,133 @@
+  let emchrs = document.querySelector(".email-checker-show");
+  if (!regex.test(email.value) && email.value != "") {
+    let emchr = document.querySelector(".email-checker");
+    emchr.classList.add("email-checker-show");
+  } else if (emchrs) {
+    emchrs.classList.remove("email-checker-show");
+  }
+  const ckems = document.querySelector(".check-email-show");
+  if (ckems) {
+    ckems.classList.remove("check-email-show");
+  }
+  if (email.value != "") {
+    removeEmailRequest();
+  }
+}
+email.addEventListener("focusout", emailChecker);
+
+function pwRequest(event) {
+  //비밀번호를 입력해 주세요
+  if (pw.value == "") {
+    let pwrq = document.querySelector(".pw-request");
+    pwrq.classList.add("pw-request-show");
+  }
+  let ckpws = document.querySelector(".check-pw-show");
+  if (ckpws) {
+    ckpws.classList.remove("check-pw-show");
+  }
+}
+
+function removePwRequest(event) {
+  let pwrqs = document.querySelector(".pw-request-show");
+  if (pwrqs) {
+    pwrqs.classList.remove("pw-request-show");
+  }
+}
+
+pw.addEventListener("focusout", pwRequest);
+pw.addEventListener("focusin", removePwRequest);
+
+function loginFunction(event) {
+  //이메일(비밀번호)을 확인해주세요.
+  if (email.value == "test@codeit.com" && pw.value == "codeit101") {
+    window.location.href = "folder.html";
+  } else if (email.value != "" && pw.value != "") {
+    if (regex.test(email.value)) {
+      const checkEmail = document.querySelector(".check-email");
+      const checkPw = document.querySelector(".check-pw");
+      checkEmail.classList.add("check-email-show");
+      checkPw.classList.add("check-pw-show");
+      let emchrs = document.querySelector(".email-checker-show");
+      if (emchrs) {
+        emchrs.classList.remove("email-checker-show");
+      }
+      removePwRequest();
+    } else {
+      emailChecker();
+    }
+  } else if (email.value == "") {
+    emailRequest();
+  } else if (pw.value == "") {
+    pwRequest();
+    emailChecker();
+  }
+}
+login.addEventListener("click", function (event) {
+  event.preventDefault();
+  loginFunction();
+});
+email.addEventListener("keypress", function (event) {
+  if (event.key == "Enter") {
+    event.preventDefault();
+    loginFunction();
+  }
+});
+pw.addEventListener("keypress", function (event) {
+  if (event.key == "Enter") {
+    event.preventDefault();
+    loginFunction();
+  }
+});
+//눈모양 아이콘
+function pwVisibility(event) {
+  if (pw.type == "password") {
+    pw.type = "text";
+    eyeoff.removeChild(eyeoff.firstChild);
+    const eyeonImg = document.createElement("img");
+    eyeonImg.setAttribute("src", "signin/eye-on.png");
+    eyeoff.prepend(eyeonImg);
+  } else {
+    pw.type = "password";
+    eyeoff.removeChild(eyeoff.firstChild);
+    const eyeoffImg = document.createElement("img");
+    eyeoffImg.setAttribute("src", "signin/eye-off.png");
+    eyeoff.prepend(eyeoffImg);
+  }
+}
+eyeoff.addEventListener("click", function (event) {
+  event.preventDefault();
+  pwVisibility();
+});
```

Comment:
함수를 조금 더 모듈러 하게 작성해봐야 할것 같아요! 아직 시간 있으니 다음과 같은 형태로 구조를 변경해보는건 어떨까요? 

```
/**
 * HTML DOM Elements
 */
const emailInput = document.querySelector('#email-input')
const passwordInput = document.querySelector('#password-input')
const loginButton = document.querySelector('#login-button')
const showPasswordButton = document.querySelector(".eyeoff");
/**
 * UI Event Listeners
 */

/**
 * Utility Logics
 */
function isEmailFormat(emailString) {
  const regex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
  return regex.test(emailString)
}

function isEmpty(text) {
  return text.length === 0
}

/**
 * Validates a user's login information.
 *
 * This function checks if the provided login information matches a predefined valid user schema.
 * It compares both email and password from the loginInfo object with the valid user credentials.
 *
 * @param {Object} loginInfo - An object containing the user's login information.
 * @param {string} loginInfo.email - The email address of the user.
 * @param {string} loginInfo.password - The password of the user.
 * @returns {boolean} - Returns `true` if the email and password match the valid user schema, otherwise returns `false`.
 */
function isValidUser(loginInfo) {
  const validUserSchema = { email: "codeit@codeit.com", password: "password" }

  const { email, password } = loginInfo
  
  if (email !== validUserSchema.email) {
    return false
  }

  if (password !== validUserSchema.password) {
    return false
  }

  return true
}

function validateUserLoginInfo() {
  const email = emailInput.value; 

  const isEmailEmpty = isEmpty(email)
  if (isEmailEmpty) {
    toggleError(emailErrorMessage, "이메일을 입력하세요")
    return false
  }

  const isValidEmailFormat = isEmailFormat(email)
  if (!isValidEmailFormat) {
    toggleError(emailErrorMessage, '잘못된 이메일 유형입니다')
    return false
  }

  const password = passwordInput.value;
  const isPasswordEmpty = isEmpty(password)
  if (isPasswordEmpty) {
    toggleError(passwordErrorMessage, "비밀번호를 입력하세요")
    return false
  }

  return true
}
/**
 * Service Logic
 */
function handleLogin(e) {
  const isValidInput = validateUserLoginInfo()
  if (!isValidInput) {
    return false
  }

  // login 로직 실행
  const user = isValidUser({ email, password });
  
  if (!user) {
    toggleError(emailErrorMessage, "이메일을 확인하세요")
    toggleError(passwordErrorMessage, "비밀번호를 확인하세요")
    return
  }

  location.href = "로그인 성공 후 이동 될 엔드포인트"
}
```